Assets/Scripts/UI/DeathScreenHelper.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Events;
+using UnityEngine.InputSystem;
+
+public class DeathScreenHelper : MonoBehaviour {
+	private static readonly int showHash = Animator.StringToHash("Show");
+	private static readonly int restartHash = Animator.StringToHash("Restart");
+	private static readonly int humanDeathHash = Animator.StringToHash("HumanDeath");
+	private static readonly int wolfDeathHash = Animator.StringToHash("WolfDeath");
+	private static readonly int regularHash = Animator.StringToHash("Regular");
+	private static readonly int piercingHash = Animator.StringToHash("Piercing");
+
+	[SerializeField] private AnimationClip animationClip;
+	[SerializeField] private SceneHelper sceneHelper;
+	[SerializeField] private Animator bodyAnimator;
+
+	private Health playerHealth;
+	private Transformation playerTransformation;
+	private Animator animator;
+	private CheckpointManager checkpointManager;
+
+	private float screenDuration = 0f;
+
+	private void Awake() {
+		screenDuration = animationClip.length;
+
+		animator = GetComponent<Animator>();
+
+		GameObject go = GameObject.FindWithTag("Player");
+		playerHealth = go.GetComponent<Health>();
+		playerTransformation = go.GetComponent<Transformation>();
+	}
+
+	private void Start() {
+		checkpointManager = FindObjectOfType<CheckpointManager>();
```

Comment:
Maybe add a tag to the checkpoint manager and use that to find it instead to avoid having to search through the whole scene for one.